### PR TITLE
CMR=8672: Added generic association functionality for services.

### PR DESCRIPTION
**Pull request by sxu123**

### CODE REVIEW

File: indexer-app/src/cmr/indexer/data/concepts/association_util.clj [hunk 1]
```diff
@@ -3,56 +3,61 @@
...
+  "Remove the concept info related to for-concept-id, convert the concept info that's associated
+  with the for-concept-id into standard concept-id revision-id."
+  [association for-concept-id]
+  (let [source-id (or (:source-concept-identifier association)
+                      (:service-concept-id association)
+                      (:tool-concept-id association)
+                      (:variable-concept-id association))
+        ;;There are no service-revision-id tool-revision-id and variable-revision-id.
+        ;;because the service/tool/variable collection associations don't allow revision id
+        ;;for serivces, tools and variables.
```
Comment (jaybarra) [inline indexer-app/src/cmr/indexer/data/concepts/association_util.clj:19]:
```suggestion
        ;; There are no service-revision-id, tool-revision-id or variable-revision-id.
```

File: search-app/src/cmr/search/results_handlers/results_handler_util.clj [hunk 2]
```diff
@@ -1,65 +1,30 @@
...
-                     (seq (util/remove-nils-empty-maps-seqs
-                           (mapv :concept-id associations))))}
-    {concept-key (or (seq (util/remove-nils-empty-maps-seqs
-                           (mapv :associated-concept-id associations)))
-                     (seq (util/remove-nils-empty-maps-seqs
-                           (mapv :concept-id associations))))}))
+  [concept-key associations]
+  ;;All the associations, generic or service/tool/variable associations,
+  ;;have been converted in indexer-app/src/cmr/indexer/data/concepts/association_util.clj
+  ;;to only contain concept-id and revision-id.
```
Comment (eereiter) [inline search-app/src/cmr/search/results_handlers/results_handler_util.clj:12]: Please remove ";;to only contain concept-id and revision-id."  This isn't true and it also isn't needed.

Comment (jaybarra) [inline search-app/src/cmr/search/results_handlers/results_handler_util.clj:12]: Is this comment necesssary? An absolute path to the location of where conversion happens may drift out of sync with any future work as well.

**Review by jaybarra — APPROVED** [review]

**Review by eereiter — APPROVED** [review]

File: search-app/src/cmr/search/results_handlers/results_handler_util.clj (hunk 2 above)
Comment (sxu123) [reply search-app/src/cmr/search/results_handlers/results_handler_util.clj:12]: Sorry for being inaccurate. I just meant to show that all the ids and revision-ids are converted to concept-id and revision-id.  Planning to remove all these comments once the review is done.

**Pull request merged**